spec/support/features/session_helper.rb

```diff
@@ -308,23 +308,27 @@ def sign_in_with_totp_enabled_user
 
     def acknowledge_and_confirm_personal_key(js: true)
       extra_characters_get_ignored = 'abc123qwerty'
-      code_words = []
 
-      page.all(:css, '[data-personal-key]').map do |node|
-        code_words << node.text
-      end
-
-      button_text = t('forms.buttons.continue')
+      code_words = page.all('.separator-text__code').map(&:text)
 
-      click_on button_text, class: 'personal-key-continue' if js
+      click_acknowledge_personal_key
 
-      fill_in 'personal_key', with: code_words.join.downcase + extra_characters_get_ignored
-
-      find_all('.personal-key-confirm', text: button_text).first.click
+      if js
+        page.find(':focus').fill_in with: code_words.join.downcase + extra_characters_get_ignored
+        if page.current_path == idv_personal_key_url
+          click_on t('forms.buttons.continue'), class: 'personal-key-confirm'
+        else
+          click_submit_default
+        end
```

Comment:
@anniehirshman-gsa Curious for your thoughts: The "Form Steps" application we created for document capture was implemented so that the button would say "Continue" up 'til the last step, then say "Submit". Using that same React implementation on the personal key step, I noticed it doesn't work the same way. In the current non-React implementation, we always say "Continue", even at this very last step.

![image](https://user-images.githubusercontent.com/1779930/164724042-b2980857-8d01-4a01-a843-5df1a04cab6a.png)

I could see two options to reconcile this:

- Always show "Continue" in the React implementation, rather than switching to "Submit" for the last step
- Or, switch to "Submit" for this personal key confirmation modal

Reply:
Thanks for looping me in and sorry for the delay! I could go either way on this because really submitting/verifying anything new here, they're just confirming they have the key we gave them. I kind of prefer the idea of only showing "Submit" for the big moments where they are submitting information that could really pass/fail.

So on that note, I _have_ been thinking we should change "Continue" to "Submit" on the Review information and Phone steps... but that's outside the scope of this, right? :wink:

Reply:
Gotcha, thanks. From a technical implementation then, it sounds like maybe we want to change this to something more of an explicit opt-in ("show Submit vs. Continue") as opposed to automatically switching based on it being the last step, which is how it's currently implemented.

Reply:
For posterity, this was addressed in #6280.